File: backend-app/src/main/kotlin/altszama/app/order/OrderState.kt
package altszama.app.order


enum class OrderState {
  CREATED,
  ORDERING,
  ORDERED,
  DELIVERED,
  REJECTED;

  fun getByName(name: String): OrderState {
    return when (name) {
      "CREATED" -> CREATED
      "ORDERING" -> ORDERING
      "ORDERED" -> ORDERED
      "DELIVERED" -> DELIVERED
      "REJECTED" -> REJECTED
      else -> CREATED
    }
  }
}

File: frontend-angular/src/app/views/orders/lib/orderState.ts
import {OrderEntryDto} from "../../../../frontend-client";
import OrderStateEnum = OrderEntryDto.OrderStateEnum;

export function orderStateToCaption(orderState: OrderStateEnum): string {
  switch (orderState) {
    case "CREATED":
      return "OPEN"
    case "ORDERING":
      return "ORDERING NOW"
    case "ORDERED":
      return "WAITING FOR DELIVERY"
    case "DELIVERED":
      return "DELIVERED"
    case "REJECTED":
      return "REJECTED"
  }
}
